src/main/java/duke/parser/Parser.java

```diff
@@ -0,0 +1,194 @@
+            case FIND:
+                taskList.find(input[1]);
+                break;
+            default:
+                Task task = getTask(input, type);
+                if (task != null) {
+                    taskList.add(task);
+                }
+            }
+        } catch (IndexOutOfBoundsException e) {
+            ui.echo("Duke.Main.Duke has noticed that the number you provided does not \nmatch the number of task you have." +
+                    "\nPlease enter a valid task number!");
+        } catch (NumberFormatException e) {
+            ui.echo("OOPS!!! Please enter a valid task number!");
+        }
+    }
+
+    /**
+     * Takes in the given user input and return a newly created task specified by the user.
+     * Catches any invalid command and reply appropriately.
+     * @param userInput User command.
+     * @param type Type of task to be created.
+     * @return A new task specified by the user.
+     */
+    private Task getTask(String[] userInput, Type type) {
+        Task task = null;
+
+        try {
+            if (userInput.length == 1) {
+                throw new DukeException("Error");
+            }
+
+            String[] strings = userInput[1].split("/");
+
+            if (strings.length == 1 && !userInput[0].equals("todo")) {
+                throw new ArrayIndexOutOfBoundsException();
+            }
+
+            switch (type) {
+            case DEADLINE:
+                task = new Deadline(strings[0], LocalDate.parse(strings[1].substring(3)));
+                break;
+            case EVENT:
+                task = new Event(strings[0], LocalDate.parse(strings[1].substring(3)));
+                break;
+            case TODO:
+                task = new Todo(userInput[1].strip());
+                break;
+            }
+
+            if (task == null || userInput[1].strip().isEmpty()) {
+                throw new DukeException("Error in task");
+            }
+            return task;
+
+        } catch (DukeException e) {
+            if (isCommandRecognized(userInput[0])) {
+                ui.echo("☹ OOPS!!! The description of a " + userInput[0] + " cannot be empty.");
+            } else if (e.getMessage().equals("Error in task") || e.getMessage().equals("Error")) {
+                ui.echo("☹ OOPS!!! I'm sorry, but I don't know what that means :-(");
+            }
+        } catch (ArrayIndexOutOfBoundsException e) {
+            ui.echo("☹ OOPS!!! Description is empty or invalid timeframe!");
+        } catch (DateTimeParseException e) {
+            ui.echo("☹ OOPS!!! Duke.Main.Duke could not understand given date!"
+                    + " Please enter in yyyy-mm-dd format!");
+        }
+        return task;
+    }
+
+    /**
+     * Returns a boolean depending on Duke understanding the given command.
+     * @param input User command.
+     * @return True if Duke understands the command, false otherwise.
+     */
+    private boolean isCommandRecognized(String input) {
```

Comment:
Perhaps this method could be better named, since it detects four specific types of commands rather than all commands

Reply:
You are right, thank you for pointing that out! 😄